diff --git a/node_modules/@beyonk/sapper-httpclient/lib/api/index.js b/node_modules/@beyonk/sapper-httpclient/lib/api/index.js
index a64a0e7..d76d088 100644
--- a/node_modules/@beyonk/sapper-httpclient/lib/api/index.js
+++ b/node_modules/@beyonk/sapper-httpclient/lib/api/index.js
@@ -47,6 +47,8 @@ class Api extends Behaviours {
     this.config = {
       endpoint: null,
       method: 'get',
+      contentType: 'application/json',
+      accept: 'application/json',
       payload: null,
       query: null,
       headers: {},
@@ -80,12 +82,12 @@ class Api extends Behaviours {
         cors: true,
         credentials: 'include',
         headers: Object.assign(
-          { Accept: 'application/json' },
-          hasPayload ? { 'Content-Type': 'application/json' } : {},
+          { Accept: this.config.accept },
+          hasPayload ? { 'Content-Type': this.config.contentType } : {},
           this.config.headers
         )
       },
-      hasPayload ? { body: JSON.stringify(this.config.payload) } : {},
+      hasPayload ? { body: this._formatPayload() } : {},
       this.config.overrides
     )
 
@@ -93,8 +95,8 @@ class Api extends Behaviours {
     const ep = this.config.query ? `${endpoint}?${this.config.query}` : `${endpoint}`
 
     try {
-      const { httpStatus, json } = await this._doQuery(1, client, ep, options)
-      return fn ? fn(json, httpStatus) : json
+      const { httpStatus, res } = await this._doQuery(1, client, ep, options)
+      return fn ? fn(res, httpStatus) : res
     } catch (e) {
       return this.handle(e, this.ctx)
     } finally {
@@ -120,17 +122,13 @@ class Api extends Behaviours {
       const r = await client(endpoint, options)
 
       if (r.status >= 200 && r.status < 400) {
-        let json
+        let res
 
         if (this._hasContent(r)) {
-          try {
-            json = await r.json()
-          } catch (e) {
-            console.error('Unable to parse response json', e.message)
-          }
+          res = await this._parseResponse(r)
         }
 
-        return { httpStatus: r.status, json }
+        return { httpStatus: r.status, res }
       }
 
       let content = ''
@@ -152,6 +150,40 @@ class Api extends Behaviours {
     }
   }
 
+  _formatPayload () {
+    switch (this.config.contentType) {
+      case 'application/json':
+        return JSON.stringify(this.config.payload)
+      case 'application/x-www-form-urlencoded':
+        return new URLSearchParams(Object.entries(this.config.payload))
+      default:
+        throw new HttpError("Supplied content type not recognized!", this.config.contentType)
+    }
+  }
+
+  async _parseResponse (response) {
+    switch (this.config.accept) {
+      case 'application/json':
+        try {
+          return await response.json()
+        } catch (e) {
+          console.error('Unable to parse response json', e.message)
+        }
+      break;
+      case 'image/*':
+        try {
+          const blob = await response.blob()
+          return URL.createObjectURL(blob)
+        } catch (e) {
+          console.error('Unable to parse response image', e.message)
+        }
+      break;
+      default:
+        console.error('Supplied response type not recognized!', this.config.accept)
+    }
+
+  }
+
   context (ctx) {
     if (ctx.fetch) {
       this.client = ctx.fetch
@@ -160,6 +192,26 @@ class Api extends Behaviours {
     return this
   }
 
+  contentType (contentType) {
+    this.config.contentType = contentType
+    return this
+  }
+
+  formEncoded () {
+    this.config.contentType = 'application/x-www-form-urlencoded'
+    return this
+  }
+
+  accept (accept) {
+    this.config.accept = accept
+    return this
+  }
+
+  acceptImage () {
+    this.config.accept = 'image/*'
+    return this
+  }
+
   override (override) {
     this.config.overrides = override
     return this
diff --git a/node_modules/@beyonk/sapper-httpclient/lib/entrypoint.js b/node_modules/@beyonk/sapper-httpclient/lib/entrypoint.js
index dfc7f7d..7eb6306 100644
--- a/node_modules/@beyonk/sapper-httpclient/lib/entrypoint.js
+++ b/node_modules/@beyonk/sapper-httpclient/lib/entrypoint.js
@@ -7,7 +7,7 @@ function configure (options) {
   config = options
 }
 
-function create () {
+export function create () {
   if (!config) { throw new Error('Api client must be configured') }
   return new Api(config)
 }
diff --git a/node_modules/@beyonk/sapper-httpclient/lib/error-handlers/bad-request.js b/node_modules/@beyonk/sapper-httpclient/lib/error-handlers/bad-request.js
new file mode 100644
index 0000000..c51d0a6
--- /dev/null
+++ b/node_modules/@beyonk/sapper-httpclient/lib/error-handlers/bad-request.js
@@ -0,0 +1,12 @@
+import {
+    BadRequestError
+} from '../errors.js'
+
+const BadRequestMixin = superclass => class BadRequest extends superclass {
+    badRequest (fn) {
+        this.handlers[BadRequestError.name] = fn
+        return this
+    }
+}
+
+export { BadRequestMixin }
diff --git a/node_modules/@beyonk/sapper-httpclient/lib/errors.js b/node_modules/@beyonk/sapper-httpclient/lib/errors.js
index b32a47c..3f38a4b 100644
--- a/node_modules/@beyonk/sapper-httpclient/lib/errors.js
+++ b/node_modules/@beyonk/sapper-httpclient/lib/errors.js
@@ -5,6 +5,9 @@ class HttpError extends Error {
   }
 }
 
+class BadRequestError extends HttpError {
+}
+
 class AccessDeniedError extends HttpError {
 }
 
@@ -36,6 +39,7 @@ class TooManyRequestsError extends HttpError {
 }
 
 const mapping = {
+  400: BadRequestError,
   401: AccessDeniedError,
   402: PaymentRequiredError,
   403: ForbiddenError,
@@ -54,6 +58,7 @@ function byCode (code) {
 
 export {
   byCode,
+  BadRequestError,
   AccessDeniedError,
   PaymentRequiredError,
   ForbiddenError,
