diff --git a/node_modules/@beyonk/sapper-httpclient/lib/api/api.js b/node_modules/@beyonk/sapper-httpclient/lib/api/api.js
index 81c785e..40c1089 100644
--- a/node_modules/@beyonk/sapper-httpclient/lib/api/api.js
+++ b/node_modules/@beyonk/sapper-httpclient/lib/api/api.js
@@ -21,6 +21,7 @@ class ApiBase {
     this.config = {
       endpoint: null,
       method: 'get',
+      contentType: 'application/json',
       payload: null,
       query: null,
       headers: {},
@@ -55,11 +56,11 @@ class ApiBase {
         credentials: 'include',
         headers: Object.assign(
           { Accept: 'application/json' },
-          hasPayload ? { 'Content-Type': 'application/json' } : {},
+          hasPayload ? { 'Content-Type': this.config.contentType } : {},
           this.config.headers
         )
       },
-      hasPayload ? { body: JSON.stringify(this.config.payload) } : {},
+      hasPayload ? { body: this._formatPayload() } : {},
       this.config.overrides
     )
 
@@ -126,6 +127,17 @@ class ApiBase {
     }
   }
 
+  _formatPayload () {
+    switch (this.config.contentType) {
+      case 'application/json':
+        return JSON.stringify(this.config.payload)
+      case 'application/x-www-form-urlencoded':
+        return new URLSearchParams(Object.entries(this.config.payload))
+      default:
+        throw new HttpError("Supplied content type not recognized!", this.config.contentType)
+    }
+  }
+
   context (ctx) {
     if (ctx.fetch) {
       this.client = ctx.fetch
@@ -134,6 +146,16 @@ class ApiBase {
     return this
   }
 
+  contentType (contentType) {
+    this.config.contentType = contentType
+    return this
+  }
+
+  formEncoded () {
+    this.config.contentType = 'application/x-www-form-urlencoded'
+    return this
+  }
+
   override (override) {
     this.config.overrides = override
     return this
